onnxruntime/core/providers/cpu/tensor/upsample_antialias.h

```diff
@@ -0,0 +1,769 @@
+// Copyright c Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <algorithm>
+#include <cmath>  // for round
+#include <vector>
+#include "core/framework/tensor.h"
+#include "gsl/span"
+#ifndef SHARED_PROVIDER
+#include "core/framework/op_kernel.h"
+#endif
+#include "core/providers/cpu/tensor/upsamplebase.h"
+
+namespace onnxruntime {
+
+namespace ConstValue {
+constexpr int32_t mag_factor = 1 << (22 - 1);
+}
+
+namespace {
+const uint8_t* GetLookupTableShared() {
+  // initialized once
+  static const auto* lookup_table = []() {
+    // if we have already initialized the lookup table, just return
+    // ideally we could have a global lookup table, but that account for too much space.
+    /* Handles values form -640 to 639. */
+    static uint8_t table[1280] = {0};
+    // if we have already initialized the lookup table, just return.
+    /* Handles values form -640 to 639. */
+    if (table[1279] == 255) {
+      return table;
+    }
```

Comment:
This is no longer possible as the static variable will only be initialized once.